src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -333,14 +339,13 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 				return true;
 			}
 			case VSCodeOscPt.CommandLine: {
-				let commandLine: string;
-				if (args.length >= 1 || args.length <= 2) {
-					commandLine = deserializeMessage(args[0]);
-				} else {
-					commandLine = '';
+				const commandLine = deserializeMessage(command);
+				const isTrusted = args[0] === this._nonce;
+				if (commandLine) {
+					this._createOrGetCommandDetection(this._terminal).setCommandLine(commandLine, isTrusted);
 				}
-				this._createOrGetCommandDetection(this._terminal).setCommandLine(commandLine, args[1] === this._nonce);
-				return true;
+				// A nonce may or may not be provided. If it is, then it should be the last argument.
+				return isTrusted ? args.length === 1 : args.length === 0;
```

Comment:
We want to return true so xterm.js knows we handed this sequence, otherwise a warning would show in the console